src/components/ToggleButtonHeader.tsx

```diff
@@ -0,0 +1,63 @@
+import React from 'react';
+import { useGlobalState } from '../CustomHooks/useGlobalState';
+import { useDispatch } from '../CustomHooks/useDispatch';
+import { Todo } from '../types/Todo';
+import { patchTodoFromStorage } from '../api/todos';
+import classNames from 'classnames';
+
+export const ToggleButtonHeader: React.FC = () => {
+  const { todos } = useGlobalState();
+  const dispatch = useDispatch();
+
+  const isAllTodosCompleted = todos.every(todo => todo.completed === true);
+
+  const promiseToggleCompleted = ({ id, completed, title }: Todo) => {
+    return new Promise<void>(resolve => {
+      const data = {
+        completed: completed,
+        title: title,
+      };
+
+      patchTodoFromStorage(id, data);
+      dispatch({ type: 'patchTodo', payload: { id, data } });
+
+      resolve();
+    });
+  };
+
+  const handleCompleteAllTodos = async () => {
+    let togglePromises;
+
+    if (isAllTodosCompleted) {
+      togglePromises = todos.map(todo => {
```

Comment:
you don't use return value of `Promise.allSettled` call. Also, you can do this in a common function (not async function). So just use forEach and toggle status for each todo